tempesta_fw/http_parser.c

```diff
@@ -2625,59 +2643,137 @@ __parse_http_date(TfwHttpMsg *hm, unsigned char *data, size_t len)
 
 	__FSM_STATE(I_DateHourSP) {
 		if (likely(c == ' '))
-			__FSM_I_MOVE(I_DateHour);
+			__FSM_I_MOVE(I_DateHourBeg);
 		return CSTR_NEQ;
 	}
 
+	__FSM_STATE(I_DateHourBeg) {
+		size_t n;
+		__fsm_sz = __data_remain(p);
+		n = __fsm_sz < 3 ? __fsm_sz: 3;
```

Comment:
That's just `n = __fsm_sz & 3`. Please avoid conditions in the parser - it's too hot.

Reply:
Ehm... no. For example, if `__fsm_sz` is 4, the statement `n = __fsm_sz & 3` will result in `0` rather than expected `3`.

Reply:
Yeah, my bad